src/discussion/discussion.controller.ts

```diff
@@ -0,0 +1,94 @@
+import { CreateChildCommentDto } from './dto/add-child-comment.dto';
+import { CreateParentCommentDto } from './dto/add-parent-comment.dto';
+import { CreateDiscussionDto } from './dto/create-discussion.dto';
+import { UpdateCommentDto } from './dto/update-comment.dto';
+
+@ApiBearerAuth('access-token')
+@UseGuards(AuthenticationGuard)
+@ApiTags('Discussions')
+@Controller('discussion')
+export class DiscussionController {
+    constructor(private readonly discussionService: DiscussionService){}
+
+    @Post('')
+    @ApiOperation({summary: 'Start a new discussion'})
+    async startDiscussion(
+        @Body() dto: CreateDiscussionDto
+    ): Promise<Ok<Discussion>>{
+        const discussion = await this.discussionService.startDiscussion(dto)
+        return ZaLaResponse.Ok(discussion, "Discussion started", "201")
+    }
+
+    @IdCheck('discussionId')
+    @Get('/:discussionId')
+    @ApiOperation({summary: 'Get a single discussion'})
+    async getSingleDiscussion(
+        @Param('discussionId') discussionId: string
+    ): Promise<Ok<Discussion>>{
+        const discussion = await this.discussionService.getSingleDisussion(discussionId) 
+        return ZaLaResponse.Ok(discussion,'Ok','200')
+    }
+
+    @IdCheck('apiId')
+    @Get('/api/:apiId')
+    @ApiOperation({summary:'Get all discussions of an api'})
+    async getAllDiscussions(
+        @Param('apiId') apiId: string
+    )
+    : Promise<Ok<Discussion[]>>{
+        const discussions = await this.discussionService.getAllDiscusionsOfAnApi(apiId)
+        return ZaLaResponse.Ok(discussions, 'Ok', '200')
+    }
+
+    @IdCheck('profileId')
+    @Post('add-parent-comment/:profileId')
+    @ApiOperation({summary: 'Create a parent comment'})
+    async addParentDiscussion(
+        @Param('profileId') profileId: string,
+        @Body() dto: CreateParentCommentDto
+    ){
+        const parentComment= await this.discussionService.addParentComment(profileId, dto)
+        return ZaLaResponse.Ok(parentComment, 'Ok', '201')
+    }
+
+    @IdCheck('profileId','commentId')
+    @Patch('/:profileId/:commentId')
+    @ApiOperation({summary: 'Edit a comment'})
+    async editComment(
+        @Param('profileId') profileId: string,
+        @Param('commentId') commentId: string,
+        @Body() dto: UpdateCommentDto
+    ):Promise<Ok<string>>{
+        await this.discussionService.editComment(profileId, commentId, dto)
+        return ZaLaResponse.Ok('Comment updated', 'Ok','200')
+    }
+
+    @IdCheck('profileId','commentId')
+    @Post('/add-child-comment/:profileId/:commentId')
+    @ApiOperation({summary:'Add a child comment to a parent comment'})
+    async addChildComment(
+        @Param('profileId') profileId: string,
+        @Param('commentId') commentId: string,
+        @Body() dto: CreateChildCommentDto
+    ){
+        const childComment = await this.discussionService.addChildComment(profileId, commentId, dto)
+        return ZaLaResponse.Ok(childComment, 'Ok', '201')
+    }
```

Comment:
Are you supposed to go 3 steps deep? A post has a title and body, and that post has comments under it. I think it should stop  here

Reply:
I don't understand your comment. Can you please explain it a little bit more